packages/common/src/store/pages/profile/reducer.ts

```diff
@@ -321,51 +328,93 @@ const actionsMap = {
     action: FetchTopTagsFailedAction
   ) {
     return updateProfile(state, action, { topTagsStatus: Status.ERROR })
+  },
+  [SET_IS_INITIAL_FETCH_AFTER_SSR](
+    state: ProfilePageState,
+    action: SetIsInitialFetchAfterSSRAction
+  ) {
+    return {
+      ...state,
+      isInitialFetchAfterSsr: action.isInitialFetchAfterSsr
+    }
   }
 }
 
 const feedLineupReducer = asLineup(feedPrefix, feedReducer)
 const tracksLineupReducer = asLineup(tracksPrefix, tracksReducer)
 
-const reducer = (
-  state = initialState,
-  action:
-    | ProfilePageAction
-    | LineupActions<Track>
-    | LineupActions<Track | Collection>
-) => {
-  const { currentUser, entries } = state
+const buildInitialState = (ssrPageProps?: SsrPageProps) => {
+  // If we have preloaded data from the server, populate the initial
+  // page state with it
+  if (ssrPageProps?.user) {
+    const { user } = ssrPageProps
+    const lowerHandle = user.handle.toLowerCase()
 
-  const profileHandle =
-    ('handle' in action && action.handle?.toLowerCase()) ?? currentUser
-  if (!profileHandle) return state
+    return {
+      ...initialState,
+      currentUser: lowerHandle,
+      entries: {
+        [lowerHandle]: {
+          ...initialProfileState,
+          handle: lowerHandle,
+          userId: decodeHashId(user.id)
+        }
+      },
+      isInitialFetchAfterSsr: true
+    }
+  }
+  return initialState
+}
+
+const reducer =
+  (ssrPageProps?: SsrPageProps) =>
+  (
+    state: ProfilePageState,
+    action:
+      | ProfilePageAction
+      | LineupActions<Track>
+      | LineupActions<Track | Collection>
+  ) => {
+    if (!state) {
+      // @ts-ignore
+      state = buildInitialState(ssrPageProps)
```

Comment:
cool, is there a way to not ts-ignore, and also checking if this is the same pattern as track page?